sdcm/sct_events.py

```diff
@@ -186,15 +213,37 @@ def __str__(self):
 
 
 class DbEventsFilter(SystemEvent):
-    def __init__(self, type, line=None, node=None):  # pylint: disable=redefined-builtin
+    def __init__(self, type, line=None, node=None,  # pylint: disable=redefined-builtin,too-many-arguments
+                 guaranteed_delivery: bool = False, guaranteed_delivery_options: Optional[dict] = None):
         super(DbEventsFilter, self).__init__()
         self.id = id(self)  # pylint: disable=invalid-name
         self.type = type
         self.line = line
         self.node = str(node) if node else None
         self.clear_filter = False
         self.expire_time = None
-        self.publish()
+        if not guaranteed_delivery:
```

Comment:
a bit closer, 
but you are missing one point, we want this option not on specific DbFilterEvents.
If I understand correctly the issue is we can send event before the ZMQ is fully ready to accept them.
If that's the case, we want to want until it's ready, and if they to figure out it's ready, is to send a message in and wait for it, lets do that at the beginning at `start_events_devices()` once, and not for each event.

If we still losing event while the ZMQ is already up and running and message already passed-throw, I need to understand why, before going into this resending again and again  kind of trick

Reply:
I've already tried this solution and found that even if MQ lets some events go thru, sometimes it eats next events, so it is not fully reliable, at least in test environment.
And basically i'am not sure if we have this issue on running , but what we can do is to add same logic to be run for any sctEvent, add marking events that we have resent, throw it thru longevity run and see what we will get there.

Reply:
If it happens it testing, is happens also not in testing.
and we did face issue when filter didn't worked, assuming this is was the root cause.

Let's add it to all message sending, if we have reason to believe it doesn't work, we rather receive multiple message or not get them at all.

Reply:
> If it happens it testing, is happens also not in testing.
> and we did face issue when filter didn't worked, assuming this is was the root cause.
> 
> Let's add it to all message sending, if we have reason to believe it doesn't work, we rather receive multiple message or not get them at all.

Negative consuquence of it is significant overhead in event processing. Not sure how it will playout in case of high events traffic.

Reply:
Let try it, we are not aiming on throughput of million of events, we are actually trying to minimize the amounts to a bare minimum.
we care more about not missing the critical ones. the DbFilter for example are importent, cause if they aren't in place, there would be a rain of unneeded critical event that would fail the test.

Reply:
> ig

I've run test to mesure the overhead and found following:
Non guranteed publishing of 1000 messages - 12.512795209884644
Guranteed publishing of 1000 messages - 13.74399471282959

So, not a big difference here.